usecase/implementation/src/main/java/eu/openg/guessnumberapi/usecase/implementation/CreateGameInteractor.java

```diff
@@ -2,26 +2,21 @@
 
 import eu.openg.guessnumberapi.domain.GameEntity;
 import eu.openg.guessnumberapi.gateway.api.GameEntityRepository;
-import eu.openg.guessnumberapi.gateway.api.GameIdProvider;
 import eu.openg.guessnumberapi.gateway.api.NumberGateway;
 import eu.openg.guessnumberapi.usecase.api.CreateGameUseCase;
 
 public class CreateGameInteractor implements CreateGameUseCase {
     private final NumberGateway gateway;
     private final GameEntityRepository gameEntityRepository;
-    private final GameIdProvider gameIdProvider;
 
-    public CreateGameInteractor(NumberGateway gateway, GameEntityRepository gameEntityRepository, GameIdProvider gameIdProvider) {
+    public CreateGameInteractor(NumberGateway gateway, GameEntityRepository gameEntityRepository) {
         this.gateway = gateway;
         this.gameEntityRepository = gameEntityRepository;
-        this.gameIdProvider = gameIdProvider;
     }
 
     @Override
     public int createGameAndReturnGameId() {
-        int id = gameIdProvider.getNextId();
         int generatedNumber = gateway.generateNumber();
-        gameEntityRepository.save(new GameEntity(id, 0, generatedNumber));
-        return id;
+        return gameEntityRepository.save(new GameEntity(0, 0, generatedNumber));
```

Comment:
pasikurk atskirą konstruktorių modelyje, kuriam nereiktų paduoti id ir number of guesses ir jį naudok čia